pkg/services/authn/clients/session.go

```diff
@@ -51,14 +56,26 @@ func (s *Session) Authenticate(ctx context.Context, r *authn.Request) (*authn.Id
 		return nil, authn.ErrTokenNeedsRotation.Errorf("token needs to be rotated")
 	}
 
-	return &authn.Identity{
+	ident := &authn.Identity{
 		ID:           authn.NamespacedID(authn.NamespaceUser, token.UserId),
 		SessionToken: token,
 		ClientParams: authn.ClientParams{
 			FetchSyncedUser: true,
 			SyncPermissions: true,
 		},
-	}, nil
+	}
+
+	info, err := s.authInfoService.GetAuthInfo(ctx, &login.GetAuthInfoQuery{UserId: token.UserId})
+	if err != nil {
+		if !errors.Is(err, user.ErrUserNotFound) {
+			s.log.FromContext(ctx).Error("Failed to fetch auth info", "err", err)
+		}
+		return ident, nil
+	}
+
+	ident.AuthID = info.AuthId
+	ident.AuthenticatedBy = info.AuthModule
```

Comment:
Just for clarification: when the user doesn't have external auth info (f.e. form login with username/password) then the `AuthenticatedBy` should be empty?

Reply:
Yes